core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -265,7 +265,7 @@ public void print(TaskListener listener) {
 
         @Override
         public boolean equals(Object o) {
-            return o instanceof CLICause;
+            return o != null && o.getClass() != getClass();
```

Comment:
Wouldn't this implementation give `false` for `this.equals(this)`?

Reply:
This seems like a good excuse to insert a test written with [equalsverifier](https://jqno.nl/equalsverifier/).

Reply:
@zbynek : You are right, I corrected it.
@MarkEWaite I'll give it a try.